### PR TITLE
Fixing and simplifying CI/CD release process. 

### LINKED ISSUE #402: Github-tag-actions is not stable in current CD workflow and needs rework
**Describe the bug**

When trying to deploy: https://github.com/idaholab/MontePy/actions/runs/8695002249/job/23845633498

The following error occurred: `Error: Reference update failed`.

This is not very helpful, but it does occur in [github-tag-action](https://github.com/mathieudutour/github-tag-action), which is based on another [GHA](https://github.com/anothrNick/github-tag-action).

Neither of these repos seems well maintained. I am leaning towards just replicating these functionalities and removing it as a dependency. The semver feature will be hard to replicate though.

### LINKED ISSUE #403: Cleanup Sdists
Right now a lot of excess files are included in an `sdist` for montepy, which I thought would be excluded.

```
AUTHORS  LICENSE  NOTICE.txt  PKG-INFO  README.md  demo  doc  graphics  montepy  montepy.egg-info  pyproject.toml  setup.cfg  tests
```

Probably should exclude `doc` `demo` and `graphics`. I assume `.egg-info` is always included.

### LINKED ISSUE #407: Bugs in CI/CD workflow
1. Changelog changes are checked for deploys. This should be skipped for main branch: https://github.com/idaholab/MontePy/actions/runs/9306739089/job/25616555261
2. GHA doesn't have permissions to push to main (and probably tags): https://github.com/idaholab/MontePy/actions/runs/9306739088

### PR DESCRIPTION
**Pull request by MicahGale**
# Description

This simplifies our CD process by making it more manual. This is done by:

* Removing all automatic versioning (via github-tag-action).
* Relying on an admin to update the changelog version numbers, and create a new tag.
* Relying on an admin to manually merge and push this merge. This is necessary because GH will always try to make a new commit which will make `setuptools_scm` think this is a dev release
* Verify that this is not a dev release
* 

Fixes #407, Fixes #403, Fixes #402.

# Checklist

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation (if applicable)
- [x] I have added tests that prove my fix is effective or that my feature works (if applicable)

### CODE REVIEW

**Comment by MicahGale** [comment]
This depends on #417

**Comment by MicahGale** [comment]
Workflow showing that this will not deploy if the tag isn't done properly. https://github.com/MicahGale/MontePy/pull/7

**Comment by MicahGale** [comment]
Successful run. https://github.com/MicahGale/MontePy/actions/runs/9653118535

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -140,6 +136,7 @@ jobs:
 
   changelog-test:
     runs-on: ubuntu-latest
+    if: github.ref != 'refs/heads/master'
```
Comment (tjlaboss) [inline .github/workflows/main.yml:139]: Should it be `main` for this repo?

Comment (MicahGale) [reply .github/workflows/main.yml:139]: Oh you're right.

**Review by tjlaboss — APPROVED** [review]

**Pull request merged**